libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/LargeCausalIndividualChart.tsx

```diff
@@ -0,0 +1,326 @@
+      localCausalData: undefined,
+      localCausalErrorMessage: undefined,
+      selectedPointsIndexes: [],
+      xSeries: [],
+      ySeries: []
+    });
+    this.props.onDataClick(false, undefined);
+    const datasetBarConfigOverride = await getBubblePlotData(
+      chartProps,
+      this.props.cohort,
+      this.context.jointDataset,
+      this.context.dataset,
+      this.state.isLocalCausalDataLoading,
+      this.context.requestBubblePlotData,
+      this.selectPointFromChartLargeData,
+      this.onBubbleClick
+    );
+    if (
+      datasetBarConfigOverride &&
+      !instanceOfHighChart(datasetBarConfigOverride)
+    ) {
+      this.setState({
+        bubbleChartErrorMessage: getErrorMessage(datasetBarConfigOverride),
+        isBubbleChartDataLoading: false,
+        plotData: undefined
+      });
+      return;
+    }
+    this.setState({
+      chartProps,
+      isBubbleChartDataLoading: false,
+      isBubbleChartRendered: true,
+      isRevertButtonClicked: false,
+      plotData: datasetBarConfigOverride
+    });
+  };
+
+  private updateScatterPlotData = (chartProps: IGenericChartProps): void => {
+    const datasetBarConfigOverride = getScatterOption(
+      this.state.xSeries,
+      this.state.ySeries,
+      this.state.indexSeries,
+      chartProps,
+      this.context.jointDataset,
+      this.state.selectedPointsIndexes,
+      [],
+      this.state.isLocalCausalDataLoading,
+      true,
+      false,
+      this.selectPointFromChartLargeData
+    );
+    this.setState({
+      chartProps,
+      isBubbleChartRendered: false,
+      isRevertButtonClicked: false,
+      plotData: datasetBarConfigOverride
+    });
+  };
+
+  private onBubbleClick = (
+    scatterPlotData: IHighchartsConfig,
+    xSeries: number[],
+    ySeries: number[],
+    indexSeries: number[]
+  ): void => {
+    this.setState({
+      indexSeries,
+      isBubbleChartRendered: false,
+      plotData: scatterPlotData,
+      xSeries,
+      ySeries
+    });
+  };
+
+  private onXSet = (value: ISelectorConfig): void => {
+    if (this.state.chartProps) {
+      const newProps = getNewChartProps(value, true, this.state.chartProps);
+      this.setState({ chartProps: newProps });
+    }
+  };
+
+  private onYSet = (value: ISelectorConfig): void => {
+    if (this.state.chartProps) {
+      const newProps = getNewChartProps(value, false, this.state.chartProps);
+      this.setState({ chartProps: newProps });
+    }
+  };
+
+  private selectPointFromChartLargeData = async (
+    data: IScatterPoint
+  ): Promise<void> => {
+    selectPointFromChartLargeData(
+      data,
+      this.setLocalCausalData,
+      this.toggleSelectionOfPoint,
+      this.props.onDataClick,
+      this.setTemporaryPointToCopyOfDatasetPoint,
+      this.props.telemetryHook
+    );
+  };
+
+  private setLocalCausalData = async (absoluteIndex: number): Promise<void> => {
+    this.setState({
+      isLocalCausalDataLoading: true,
+      localCausalErrorMessage: undefined
```

Comment:
Any way we can try to remove usage of undefined directly?  I thought it was discouraged in UI code based on what Ke said.  I see a lot of undefined everywhere here in the PR.